File: java/direct-policy/src/test/java/org/nhindirect/policy/PolicyValueFactory_getInstanceTest.java
package org.nhindirect.policy;

import junit.framework.TestCase;

public class PolicyValueFactory_getInstanceTest extends TestCase
{
	public void testGetInstance_assertValue()
	{
		PolicyValue<Integer> value = PolicyValueFactory.getInstance(12345);
		
		assertTrue(value.getPolicyValue() instanceof Integer);
		assertEquals(12345, value.getPolicyValue().intValue());
		assertEquals("12345", value.toString());
		assertEquals(new Integer(12345).hashCode(), value.getPolicyValue().hashCode());
		
		
		assertFalse(value.equals(null));
		assertTrue(value.equals(12345));
		assertTrue(value.equals(PolicyValueFactory.getInstance(12345)));
		
		
		assertTrue(value.equals(12345));
		assertFalse(value.equals("12345"));
	}
	
}

File: csharp/unittests/policy.tests/PolicyValueFactory_GetInstanceTest.cs
using FluentAssertions;
using Xunit;

namespace Health.Direct.Policy.Tests
{
    public class PolicyValueFactory_GetInstanceTest
    {
        [Fact]
        public void testGetInstance_assertValue()
        {
            IPolicyValue<int> value = PolicyValueFactory.GetInstance(12345);
            value.GetPolicyValue().Should().Be(12345);
            value.Should().Be(PolicyValueFactory.GetInstance(12345));

            IPolicyValue<string> value2 = PolicyValueFactory.GetInstance("12345");
            value2.GetPolicyValue().Should().Be("12345");
            value2.Should().Be(PolicyValueFactory.GetInstance("12345"));
        }
	
    }
}